asv/benchmarks.py

```diff
@@ -182,12 +184,11 @@ def iter_unique(iter):
         log.info("Discovering benchmarks")
         with log.indent():
             last_err = None
-            for env, commit_hash in itertools.product(environments, try_hashes):
+            tools = itertools.product(environments, try_hashes)
+            for env, commit_hash in tools:
```

Comment:
Probably easier, and to me at lease, more readable, to use this:

```suggestion
            for env, commit_hash in itertools.product(environments,
                                                      try_hashes):
```